seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java

```diff
@@ -79,7 +89,18 @@ public void write(@NonNull SeaTunnelRow seaTunnelRow) {
             } else {
                 fsDataOutputStream.write(rowDelimiter.getBytes());
             }
-            fsDataOutputStream.write(serializationSchema.serialize(seaTunnelRow));
+            if (seaTunnelRow.getFields().length != sinkColumnsIndexInRow.size()) {
```

Comment:
`seaTunnelRow.getFields().length != sinkColumnsIndexInRow.size()` is Imperfect.

Maybe `seaTunnelRow.getFields().length == sinkColumnsIndexInRow.size()` but the field order is different.

Reply:
Project operation should implement in transform layer, sink layer should do something more pure, IMO, if we add this logic in sink endpoint, we should produce a new row to do this work, this will lower performance.

Reply:
'is_partition_field_write_in_File 'is the configuration of the sink layer. It can't be judged at the transform layer. I tried some methods, but finally I still need to create a new line. I can't find a better way to deal with it.

Reply:
If there is a better way to propose, I will try to revise it.